Networking/Networking/Requests/RESTRequest.swift

```diff
@@ -44,7 +50,7 @@ struct RESTRequest: URLRequestConvertible {
     /// Returns a URLRequest instance representing the current REST API Request.
     ///
     func asURLRequest() throws -> URLRequest {
-        let url = try (siteURL + path).asURL()
+        let url = try (siteURL.removingSuffix("/") + Settings.basePath + wooApiVersion.path.removingPrefix("/") + path.removingPrefix("/")).asURL()
```

Comment:
I wonder if this can be an issue if `wooApiVersion` is `none` (for WPOrg requests for example) - then we'll miss a "/" in front of `path`.

Can we make this simpler by removing all surrounding "/" of each component, and join them together with a "/" separator?

```suggestion
        let components = [siteURL.removingSuffix("/"), "wp-json", wooApiVersion.path.removingPrefix("/").removingSuffix("/"), path.removingPrefix("/")]
        let url = try components.joined("/").asURL()
```

The suggested `components` list still looks a bit messy, please feel free to clean it up as needed 😅

Reply:
Updated: we will not miss the "/" in front of `path` since there's one at the end of `basePath` 😅. But it would be great if we can rewrite this in a way to make it more readable.

Reply:
I tried moving this into an extension [here](https://github.com/woocommerce/woocommerce-ios/pull/8475/commits/580cc69dd68a8e7223c7047343d03560b4d2bd04)

Kindly let me know if we can improve it further.

Reply:
Looks great! 💯